### DISCUSSION TITLE
Throwing hardware at my model - options?

### DISCUSSION THREAD

**Discussion by RibeyeOnTheBone** [coin-or/Cbc#430] [Q&A]
As a beginner I am running my COIN-BC models from Minizinc. One has just doubled in number of lines and is not completing on my laptop.

I am looking to throw hardware at it and want to experiment. It seems a simple thing to take the Minizinc docker image and run on AWS. AWS has "general", "compute optimized", "memory optimized", "accelerated computing" instance types. I am looking for rules of thumb to shorten my search - any suggestions? I cant find commentary on how to approach the problem - are there any guides?

**Comment by tosttost** [comment]
"accelerated computing" won't help as Cbc does not use the GPU.

In my experience throwing more hardware on a problem has very limited effect. All mixed integer programming is NP-hard and all relevant existing solvers (including CBC as well as commercial like CPLEX and Gurobi) use an algorithm with (at least) exponential worst case complexity. Once your model/solver combination goes that way no hardware in the world will get you out.

To make a model solve faster I would try to improve the model. How to improve a given model requires some experience, inspection of the logfile/stdout and understanding the application/model. 

In case Cbc struggles to find a feasible solution: try to provide an initial solution, even a low quality solution can help. 
Playing with the options can also speed up things.

Sorry for being very generic, but I can't fix your model in my spare time even if you provide more information...

**Reply by RibeyeOnTheBone** [reply]
Many thanks @tosttost. That has potentially saved me some hours, if not days. Not what I wanted to hear exactly, but the initial solution option is doable and I will spend time on that instead. Thanks for taking the time.
